src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -7,25 +7,24 @@ import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address
 import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
 
 type BlockCID ipld.CID
-type MessageRoot ipld.CID
-type ReceiptRoot ipld.CID
 type ChainWeight UVarint
 type ChainEpoch UVarint
+type MessageReceipt util.Bytes
 
 // On-chain representation of a block header.
 type BlockHeader struct {
     // Chain linking
     Parents          Tipset
```

Comment:
This should be `[&BlockHeader]`. There's a rabbit hole here because some existing code uses this wrongly.

Reply:
As you say, we use a lot of the `Tipset` convenience methods from `Tipset.id` elsewhere. Should i go down the rabbit hole and fix throughout or leave as is for now?

Reply:
Yes, please fix. I don't mind of those fixes involve deleting or disabling the downstream code that breaks. Implementers need this header structure to be exact and precise, but not so for other code.

Reply:
Ok. done. I feel the cached tipset methods from `tipset.id` are trivial to understand with a tipset initialized with [&BlockHeader]. If you don't agree, I can implement them all and show a Tipset constructor to really hammer home the point... Unec. i think